### PR TITLE
fix(#1143): let HttpMessageBuilder create messages with unique ids

### PR DESCRIPTION
**Pull request by tschlat**
I've updated the HttpMessageBuilder to consistently generate messages with new IDs and timestamps. As a result, messages created by the builder will always have unique identifiers and may differ in their timestamps. I haven't identified any scenarios where this change would cause any issues.

I would appreciate Christoph's thorough review and expert opinion on this matter, considering the potential for unforeseen side effects.

### CODE REVIEW

File: endpoints/citrus-http/src/main/java/org/citrusframework/http/message/HttpMessage.java [hunk 1]
```diff
@@ -555,10 +566,7 @@ public static HttpMessage fromResponseData(final String responseData) {
     }
 
     private void addQueryParam(final String name, final String value) {
-        if (!this.queryParams.containsKey(name)) {
-            this.queryParams.put(name, new LinkedList<>());
-        }
-        this.queryParams.get(name).add(value);
+        this.queryParams.computeIfAbsent(name, k->new LinkedList<>()).add(value);
```
Comment (bbortt) [inline endpoints/citrus-http/src/main/java/org/citrusframework/http/message/HttpMessage.java:569]:
```suggestion
        this.queryParams.computeIfAbsent(name, k -> new LinkedList<>()).add(value);
```

File: endpoints/citrus-http/src/test/java/org/citrusframework/http/message/HttpMessageBuilderTest.java [hunk 2]
```diff
@@ -17,29 +17,31 @@
...
-import static org.testng.Assert.assertTrue;
-
 public class HttpMessageBuilderTest {
 
     private HttpMessage message;
 
     @BeforeMethod
     public void setUp(){
         message = new HttpMessage("");
+        message.setHeader(MessageHeaders.TIMESTAMP, System.currentTimeMillis()-1);
```
Comment (bbortt) [inline endpoints/citrus-http/src/test/java/org/citrusframework/http/message/HttpMessageBuilderTest.java:44]:
```suggestion
        message.setHeader(MessageHeaders.TIMESTAMP, System.currentTimeMillis() - 1);
```

File: endpoints/citrus-http/src/test/java/org/citrusframework/http/message/HttpMessageBuilderTest.java [hunk 3]
```diff
@@ -53,8 +55,10 @@ public void testDefaultMessageHeader() {
 
         //THEN
         assertEquals(builtMessage.getHeaders().entrySet().size(), 3);
-        assertEquals(message.getHeader(MessageHeaders.ID), builtMessage.getHeader(MessageHeaders.ID));
-        assertEquals(message.getHeader(MessageHeaders.TIMESTAMP), builtMessage.getHeader(MessageHeaders.TIMESTAMP));
+        assertNotNull(message.getHeader(MessageHeaders.ID));
+        assertNotNull(message.getHeader(MessageHeaders.TIMESTAMP));
+        assertNotEquals(message.getHeader(MessageHeaders.ID), builtMessage.getHeader(MessageHeaders.ID));
+        assertNotEquals(message.getHeader(MessageHeaders.TIMESTAMP), builtMessage.getHeader(MessageHeaders.TIMESTAMP));
```
Comment (bbortt) [inline endpoints/citrus-http/src/test/java/org/citrusframework/http/message/HttpMessageBuilderTest.java:61]: I think the negative case (`forceCitrusHeaderUpdate` being `false`) deserves a test too 😉

**Review by bbortt — CHANGES_REQUESTED** [review]
as discussed, I think this change adequate. would be interesting to hear @christophd opinion, since he's clearly better with historical stuff than I am 😛 

@tschlat and some nit-picks as usual.

File: endpoints/citrus-http/src/test/java/org/citrusframework/http/message/HttpMessageBuilderTest.java (hunk 3 above)
Comment (tschlat) [reply endpoints/citrus-http/src/test/java/org/citrusframework/http/message/HttpMessageBuilderTest.java:60]: I added respective test cases at DefaultMessage and HttpMessage level.

File: endpoints/citrus-http/src/main/java/org/citrusframework/http/message/HttpMessage.java [hunk 4]
```diff
@@ -77,6 +77,17 @@ public HttpMessage(final Message message) {
         copyCookies(message);
```
Comment (bbortt) [inline endpoints/citrus-http/src/main/java/org/citrusframework/http/message/HttpMessage.java:77]: this constructor could just contain `this(message, false)`, right? just to have said it.

**Review by bbortt — APPROVED** [review]
/lgtm

**Review by christophd — COMMENTED** [review]
I do not remember exactly why this has been implemented to preserve the message ID and TIMESTAMP in HttpMessageBuilder. I just remember that the static message builder may be used several times in a kind of hierarchy and each level would then create its own IDs. I can imagine that this may cause issues when the user extracts the ID saving it to a test variable and referencing it at a later time in the test. But I can not point you exactly to this where this is being used.

In general I would rely on the unit and integration tests that we have. If those tests are green I am good to proceed with the change.

Maybe we can expose of the default for `forceCitrusHeaderUpdate` into a new setting that can be adjusted via application.properties and env variables. Then we should be good to also restore the old behavior when a user needs to do so.

WDYT?

File: endpoints/citrus-http/src/main/java/org/citrusframework/http/message/HttpMessage.java (hunk 4 above)
Comment (tschlat) [reply endpoints/citrus-http/src/main/java/org/citrusframework/http/message/HttpMessage.java:77]: That is - indeed true. I adjusted it accordingly.

File: endpoints/citrus-http/src/main/java/org/citrusframework/http/message/HttpMessage.java [hunk 5]
```diff
@@ -534,7 +548,9 @@ public static HttpMessage fromRequestData(final String requestData) {
      *
      * @param responseData The response dump to parse
      * @return The parsed dump as HttpMessage
+     * @deprecated this method is not used and will be removed in the next major release
      */
+    @Deprecated(forRemoval = true) // TODO: @Christoph is it ok to mark this as deprecated? Doesn't seem to be in use.
```
[the remark this answers is not in the document]
Comment (tschlat) [reply endpoints/citrus-http/src/main/java/org/citrusframework/http/message/HttpMessage.java]: same here

File: endpoints/citrus-http/src/main/java/org/citrusframework/http/message/HttpMessage.java [hunk 6]
```diff
@@ -505,7 +517,9 @@ public HttpMessage cookie(final Cookie cookie) {
      *
      * @param requestData The request dump to parse
      * @return The parsed dump as HttpMessage
+     * @deprecated this method is not used and will be removed in the next major release
      */
+    @Deprecated(forRemoval = true) // TODO: @Christoph is it ok to mark this as deprecated? Doesn't seem to be in use.
```
[the remark this answers is not in the document]
Comment (tschlat) [reply endpoints/citrus-http/src/main/java/org/citrusframework/http/message/HttpMessage.java]: The IDE told me, that the method is not used and after double checking this seemed true. So this was meant as a cleanup. I removed the deprecation and added a suppression plus a remark in the java doc.

**Comment by bbortt** [comment]
@tschlat thanks, I've squashed your commits.

**Pull request merged**